File: app/src/main/java/com/android/note/domain/NoteRepoImpl.java
package com.android.note.domain;

import android.graphics.Color;

import com.android.note.domain.NoteEntity;
import com.android.note.domain.NoteRepo;

import java.util.ArrayList;
import java.util.List;

/**
 * этот клас для реализации интерфейса (NoteRepo)
 * здесь реализуем методы которые указаны в NoteRepo
 */
public class NoteRepoImpl implements NoteRepo {

    //заводим массив сущьности (хранилище заметок на основе массива)
    public static List<NoteEntity> data = new ArrayList<>();

    private int counter = 0;

    //добавляем данные (создаем список)
    public NoteRepoImpl(){
        addNote(new NoteEntity(
                createRandomID(),
                "Заголовок",
                "Привет, Привет",
                Color.RED
        ));
        addNote(new NoteEntity(
                createRandomID(),
                "Название",
                "Дорогой дневник",
                Color.BLUE
        ));
        addNote(new NoteEntity(
                createRandomID(),
                "Hello",
                "Привет, Привет",
                Color.YELLOW
        ));
        addNote(new NoteEntity(
                createRandomID(),
                "Заголовок",
                "Ура",
                Color.BLACK
        ));
    }

    @Override
    public void addNote(NoteEntity noteEntity) {
        data.add(noteEntity);//добавляем заметку в список
    }

    @Override
    public List<NoteEntity> getNotes() {
        return new ArrayList<>(data);//отдаем копию списка, копию коллекции (лучше отдавать копию чтобы никто не повредил оригинальные данные)
    }

    @Override
    public void deleteNoteById(int id) {
        for (int i = 0; i < data.size(); i++) {
            if (data.get(i).getId() == id) {
                data.remove(i);
                break;
            }
        }
    }

    @Override
    public void deleteAll() {
        data.clear();
    }

    @Override
    public int createRandomID() {
        return counter++;
    }
}

File: app/src/main/java/com/android/note/data/NoteRepoImpl.kt
package com.android.note.data

import android.graphics.Color
import com.android.note.domain.NoteEntity
import com.android.note.domain.NoteRepo

/**
 * этот клас для реализации интерфейса (NoteRepo)
 * здесь реализуем методы которые указаны в NoteRepo
 */
class NoteRepoImpl : NoteRepo {
    private var counter = 0
    override fun addNote(noteEntity: NoteEntity) {
        data.add(noteEntity) //добавляем заметку в список
    }

    override fun getNotes(): List<NoteEntity> {
        return ArrayList(data) //отдаем копию списка, копию коллекции (лучше отдавать копию чтобы никто не повредил оригинальные данные)
    }

    override fun deleteNoteById(id: Int) {
        for (i in data.indices) {
            if (data[i].id == id) {
                data.removeAt(i)
                break
            }
        }
    }

    override fun deleteAll() {
        data.clear()
    }

    override fun createRandomId(): Int {
        return counter++
    }

    override fun update(changedNote: NoteEntity) {
        val id = changedNote.id //это id который мы хотим изменить

        //поиск старой заметки
        var oldNote: NoteEntity? = null
        for (i in data.indices) {
            if (data[i].id == id) { //находим нужный id
                oldNote = data[i] //получаем нужный элемент
                break
            }
        }
        if (oldNote == null) { //если не нашли заметку, то добавляем заметку
            addNote(changedNote)
        } else {
            //если заметка существует, то меняем в ней setTitle, setContent, setColor
            oldNote.title = changedNote.title
            oldNote.content = changedNote.content
            oldNote.color = changedNote.color
        }
    }

    companion object {
        //заводим массив сущьности (хранилище заметок на основе массива)
        @JvmField
        var data: MutableList<NoteEntity> = ArrayList()
    }

    //добавляем данные (создаем список)
    init {
        addNote(
            NoteEntity(
                createRandomId(),
                "Заголовок",
                "Привет, Привет",
                Color.RED
            )
        )
        addNote(
            NoteEntity(
                createRandomId(),
                "Название",
                "Дорогой дневник",
                Color.BLUE
            )
        )
        addNote(
            NoteEntity(
                createRandomId(),
                "Hello",
                "Привет, Привет",
                Color.YELLOW
            )
        )
        addNote(
            NoteEntity(
                createRandomId(),
                "Заголовок",
                "Ура",
                Color.BLACK
            )
        )
    }
}